src/main/kotlin/org/rust/cargo/runconfig/command/CargoExecutableRunLineMarkerContributor.kt

```diff
@@ -0,0 +1,27 @@
+package org.rust.cargo.runconfig.command
+
+import com.intellij.execution.lineMarker.ExecutorAction
+import com.intellij.execution.lineMarker.RunLineMarkerContributor
+import com.intellij.icons.AllIcons
+import com.intellij.psi.PsiElement
+import com.intellij.util.Function
+import org.rust.lang.core.psi.RsElementTypes.IDENTIFIER
+import org.rust.lang.core.psi.RsFunction
+import org.rust.lang.core.psi.util.elementType
+
+class CargoExecutableRunLineMarkerContributor : RunLineMarkerContributor() {
+    override fun getInfo(element: PsiElement): Info? {
+        if (element.elementType != IDENTIFIER) return null
+        val fn = element.parent as? RsFunction ?: return null
+        if (!CargoExecutableRunConfigurationProducer.isMainFunction(fn)) return null
+
+        val actions = ExecutorAction.getActions(0)
```

Comment:
https://github.com/intellij-rust/intellij-rust/commit/736a4d8d3515277fbb113b9a47e6214a95b2d342

though it does not work in this case for some reason.